### PR TITLE
Make the method `BaseAuthManager.is_authorized_custom_view` abstract

### PR DESCRIPTION
**Pull request by vincbeck**
While testing auth managers I realized that `is_authorized_custom_view` is also called when custom pages are defined in plugins. Therefore, we need to make the method `BaseAuthManager.is_authorized_custom_view` abstract so that auth managers can also work in environments with custom pages defined in plugins.

This is technically a breaking change but:
- Auth managers have been introduced recently, I am very confident nobody used the interface yet
- Documentation related to auth managers have not been released yet (will be in 2.9), hence the likelihood of having user using the interface is even less
- This is a bug fix

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/auth_manager/avp/entities.py [hunk 1]
```diff
@@ -35,8 +35,10 @@ class AvpEntities(Enum):
     # Resource types
     CONFIGURATION = "Configuration"
     CONNECTION = "Connection"
+    CUSTOM = "Custom"
     DAG = "Dag"
     DATASET = "Dataset"
+    MENU = "Menu"
```
Comment (potiuk) [inline airflow/providers/amazon/aws/auth_manager/avp/entities.py:41]: Those changes for MENU look unrelated. Or am I wrong?

**Review by o-nikolas — COMMENTED** [review]
I'm also a bit confused by the menu related changes.

Also is it possible to get unit test coverage of the plugin path? Since it was missed before

**Comment by vincbeck** [comment]
Hmmm, I can understand the MENU change looks unrelated. I can possibly split the PR in 2 if that helps. I can also a unit test to cover the plugin path

**Comment by vincbeck** [comment]
Actually, let me explain why I combined the two. The idea initially was to call `is_authorized_custom_view`in this case: https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/auth_manager/aws_auth_manager.py#L512. This case is when there is a custom menu item in the menu added by a plugin. This was my initial idea. But since recently in #37881 we added the action `MENU`, the idea was to use it as well. I can definitely do it in 2 PRs but that would mean implementing something with unit tests associated, and deleting this exact same code I just implemented in a follow-up PRs to take into consideration the action `MENU`.

Though, I am still happy to make this effort if you think that would make the review easier

**Comment by vincbeck** [comment]
I also added a unit test that loads the index page. The difference with the test being implemented in #37947, is this one is a unit test. No real API call is made against AWS. This test will fail as long as #37997 is not merged.

File: airflow/providers/amazon/aws/auth_manager/avp/entities.py (hunk 1 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/auth_manager/avp/entities.py:41]: See my comment [here](https://github.com/apache/airflow/pull/37915#issuecomment-1985892661)

**Comment by vincbeck** [comment]
@o-nikolas @potiuk any concerns on why I also included the "MENU" change in this PR? Happy to change something if needed

File: airflow/providers/amazon/aws/auth_manager/avp/entities.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/amazon/aws/auth_manager/avp/entities.py:41]: Fine for me

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
LGTM
